kedro/config/omegaconf_config.py

```diff
@@ -208,36 +228,39 @@ def load_and_merge_dir_config(
             Resulting configuration dictionary.
 
         """
-        if not Path(conf_path).is_dir():
+        # pylint: disable=too-many-locals
+
+        if not self._fs.isdir(Path(conf_path).as_posix()):
             raise MissingConfigException(
                 f"Given configuration path either does not exist "
                 f"or is not a valid directory: {conf_path}"
             )
 
         paths = [
-            Path(each).resolve()
+            Path(each)
             for pattern in patterns
-            for each in iglob(f"{str(conf_path)}/{pattern}", recursive=True)
+            for each in self._fs.glob(Path(f"{str(conf_path)}/{pattern}").as_posix())
         ]
         deduplicated_paths = set(paths)
         config_files_filtered = [
             path for path in deduplicated_paths if self._is_valid_config_path(path)
         ]
 
         config_per_file = {}
-
         for config_filepath in config_files_filtered:
             try:
-                config = OmegaConf.load(config_filepath)
+                with self._fs.open(str(config_filepath.as_posix())) as open_config:
+                    tmp_fo = io.StringIO(open_config.read().decode("utf8"))
+                    config = OmegaConf.load(tmp_fo)
```

Comment:
I think it's worth adding a comment on why this is necessary. As `fsspec` doesn't allow the file to be read as StringIO, this is a workaround to read it as a binary file and decode it back to `utf8`